src/app/vertextool/qgsvertextool.cpp

```diff
@@ -1411,7 +1413,11 @@ void QgsVertexTool::showVertexEditor()  //#spellok
 void QgsVertexTool::cleanupVertexEditor()
 {
   mLockedFeature.reset();
-  mVertexEditor.reset();
+  // do not delete immediately because vertex editor
+  // can be still used in the qt event loop
+  mVertexEditor->deleteLater();
+  mVertexEditor = nullptr;
```

Comment:
it's not necessary to do `mVertexEditor = nullptr;` if we are using QPointer (it would be needed if we used raw pointer)